core/src/test/scala/org/apache/spark/crypto/ShuffleEncryptionSuite.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.security.PrivilegedExceptionAction
+
+import org.apache.hadoop.security.{Credentials, UserGroupInformation}
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.crypto.CryptoConf._
+import org.apache.spark.crypto.CryptoStreamUtils.SPARK_CHIMERA_CONF_PREFIX
+
+private[spark] class ShuffleEncryptionSuite extends SparkFunSuite {
+  val ugi = UserGroupInformation.createUserForTesting("testuser", Array("testgroup"))
+
+  test("Test Chimera configuration conversion") {
```

Comment:
Hi @vanzin thank you for your comments. 
> It would also be super great if CHIMERA_CONF_PREFIX were defined in the Chimera library itself; not sure if that constant exists there.

Now some configurations in Chimera is not named with prefix "chimera.crypto."(e.g. chimera.secure.random.classes). In the implementation of ```toChimeraConf``` method, it finds all props prefixed with "spark.shuffle.crypto." and replaces them with "chimera.crypto.". In this way, some configurations will not take effects. I need to update the Chimera code to change all configuration in prefix of "chimera.crypto.". Another way by using current implementation of Chimera directly is that we can replace the Spark prefix "Spark.shuffle." with "chimera." instead. This will pass lots of irrelevant configurations to Chimera. I prefer the first one. Any thoughts?